File: server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/ConfiscationOfDocumentsProtocolAddRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class ConfiscationOfDocumentsProtocolAddRequest {

    @NotNull
    private final Date dateOfFill;

    @NotNull
    private final Time timeOfFill;

    @NotNull
    @NotEmpty
    private final String placeOfFill;

    @NotNull
    private final Long policeOfficerID;

    @NotNull
    private final Long carAccidentParticipant;

    @NotNull
    @NotEmpty
    private final String confiscationReason;

    @NotNull
    @NotEmpty
    private final String confiscatedDocumentsInfo;

    @NotNull
    @NotEmpty
    private final String confiscationProcessFixationMethod;

    @NotNull
    private final Long firstWitnessID;

    @NotNull
    private final Long secondWitnessID;

    @NotNull
    private final Long carAccidentEntityID;
}

File: client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/ConfiscationOfDocumentsProtocolAddRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class ConfiscationOfDocumentsProtocolAddRequest(
    val dateOfFill: Date,
    val timeOfFill: Time,
    val placeOfFill: String,
    val policeOfficerID: Long,
    val carAccidentParticipant: Long,
    val confiscationReason: String,
    val confiscatedDocumentsInfo: String,
    val confiscationProcessFixationMethod: String,
    val firstWitnessID: Long,
    val secondWitnessID: Long,
    val carAccidentEntityID: Long
)
